src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -164,11 +192,16 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
 
             SamReader reader;
             if ( samIndices == null ) {
-                reader = samReaderFactory.open(samPath);
+                reader = samReaderFactory.open(samPath, wrapper, indexWrapper);
             }
             else {
-                final SamInputResource samResource = SamInputResource.of(samPath);
-                samResource.index(samIndices.get(samCount));
+                final SamInputResource samResource = SamInputResource.of(samPath, wrapper);
+                Path indexPath = samIndices.get(samCount);
+                if (!BucketUtils.isCloudStorageUrl(samPath)) {
```

Comment:
So if the `indexPath` is on GCS and the `samPath` is not, we don't want to use an index wrapper? Can you explain why?